extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/cors/CORSFilter.java

```diff
@@ -91,30 +92,42 @@ public void handle(RoutingContext event) {
                 processRequestedHeaders(response, requestedHeaders);
             }
 
-            boolean allowsOrigin = !corsConfig.origins.isPresent() || corsConfig.origins.get().contains(origin);
+            boolean allowsOrigin = containsOnlyWildcard(corsConfig.origins)
+                    || corsConfig.origins.orElse(Collections.emptyList()).contains(origin);
 
             if (allowsOrigin) {
                 response.headers().set(HttpHeaders.ACCESS_CONTROL_ALLOW_ORIGIN, origin);
             }
 
             response.headers().set(HttpHeaders.ACCESS_CONTROL_ALLOW_CREDENTIALS, "true");
 
-            final Optional<List<String>> exposedHeaders = corsConfig.exposedHeaders;
+            if (corsConfig.exposedHeaders.isPresent()) {
+                List<String> exposedHeaders = corsConfig.exposedHeaders.orElse(Collections.emptyList());
```

Comment:
You already checked the value is present. No need for a `orElse` (I know it was preexisting but let's get rid of it).